students/ZidanLuo/session03/strformat_lab.py

```diff
@@ -0,0 +1,39 @@
+#Task One
+tp = (2, 123.4567, 10000, 12345.67)
+print("file_{:0>3d}: {:.2f}, {:.2e}, {:2e}".format(*tp))
+#Task Two
+print("file_" + '%03d' % (tp[0]) + ": " + '%.2f' % (tp[1]) + 
+	", " + '%.2e' % (tp[2]) + ', ' + '%.2e' % (tp[3]))
+#Task Three
+def formatter(tpl):
+	form_string = "the " + str(len(tpl)) + " numbers are: " + "{:d}, "*len(tpl)
+	return form_string.format(*tpl)
+
+#Task Four
+new_tpl = (4, 30, 2017, 2, 27)
+print('{:02} {:d} {:d} {:02} {:d}'.format(new_tpl[-2], 
+	new_tpl[-1], new_tpl[2], new_tpl[0], new_tpl[1]))
+
+#Task Five
+ls = ['Orange', 1.3, 'lemon', 1.1]
```

Comment:
Make sure you use meaningful names here - ls, as in the other program, can be confusing.  Something like test_list or the like.